### PR TITLE
Parse WebSocket connect protocols as array

### PR DESCRIPTION
**Pull request by JunielKatarn**
See React Native's JavasCript WebSocket spec:
```
  constructor(
    url: string,
    protocols: ?string | ?Array<string>,
    options: ?{headers?: {origin?: string, ...}, ...},
  )
...
```

Current `WebSocketModule` implementation expects an object (similar to options), which will crash if a scalar or array of strings is passed.
This change:
- Fixes the `folly::dynamic` argument handling.
- Enhances existing JavaScript-driven validation to cover this use case (currently not part of CI).
- Re-enables Desktop integration tests for both native and JavaScript coverage (may scale down due to registered bundling issues).

Requesting backport down to `0.62` due to this being a crash-inducing bug.

## Pre-merge checklist
- [ ] Assess Desktop JavaScript tests (`RNTesterIntegrationTests`).
- [ ] Add validations for `options` parameter.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6866)

### CODE REVIEW

File: .ado/jobs/desktop.yml [hunk 1]
```diff
@@ -12,16 +12,18 @@ jobs:
       - template: ../variables/vs2019.yml
       #5059 - Disable failing or intermittent tests (IntegrationTestHarness,AsyncStorage,WebSocket,Logging).
       #5265 - WebSocketModuleIntegrationTest::WebSocketModule_Ping fails for Release
+      # windesktop is no longer bundling - Disable RNTesterIntegrationTests until fixed.
```
Comment (NikoAri) [inline .ado/jobs/desktop.yml:15]: >until fixed [](start = 76, length = 11)

Please add work item number that tracks this work.

File: vnext/Desktop.IntegrationTests/RNTesterIntegrationTests.cpp [hunk 2]
```diff
@@ -109,6 +109,10 @@ TEST_CLASS (RNTesterIntegrationTests) {
     TestComponent("TimersTest");
   }
 
+  // AsyncStorage seems broken in Windows Desktop
```
Comment (NikoAri) [inline vnext/Desktop.IntegrationTests/RNTesterIntegrationTests.cpp:112]: >broken [](start = 24, length = 6)

Same comment, please add work item number that tracks this fix

**Review by NikoAri — APPROVED** [review]
:shipit:

File: .ado/jobs/desktop.yml (hunk 1 above)
Comment (JunielKatarn) [reply .ado/jobs/desktop.yml:15]: Actually,that statement seems to not apply anymore (see description).

Will remove.

**Comment by NickGerleman** [comment]
Customer ask for this is for an MSRN user right? Does this need to go back to 0.62/0.63?

File: vnext/src/IntegrationTests/WebSocketBinaryTest.js [hunk 3]
```diff
@@ -61,6 +61,10 @@ class WebSocketBinaryTest extends React.Component<{...}, State> {
   };
 
   _connect = () => {
+    // Test protocols parsing. No further use.
+    const scalarProtocolSocket = new WebSocket(this.state.url, "p0");
```
Comment (NickGerleman) [inline vnext/src/IntegrationTests/WebSocketBinaryTest.js:65]: Are these used anywhere?

Comment (NickGerleman) [reply vnext/src/IntegrationTests/WebSocketBinaryTest.js:65]: Oh, I see the comment now.

Note that creating the object here won't synchronously call the code in native. So if this triggers an action, we may see random crashes later in the test. Would be great to find some signal to wait on here.

**Comment by JunielKatarn** [comment]
> Customer ask for this is for an MSRN user right? Does this need to go back to 0.62/0.63?

Issue raised by Xbox (@aschultz).

This is a bug that will crash the app in a valid API (constructor) usage.

I believe it should be backported with ir without explicit customer ask.
Reasoning: this could crash Office by just adding the `protocols` parameter.

File: vnext/src/IntegrationTests/WebSocketBinaryTest.js (hunk 3 above)
Comment (JunielKatarn) [reply vnext/src/IntegrationTests/WebSocketBinaryTest.js:65]: Maybe asserting a correctly constructed state in the end of the test should be enough. I'll give it a try.

**Comment by NickGerleman** [comment]
@JunielKatarn could you backport this to 0.64?

**Comment by JunielKatarn** [comment]
> @JunielKatarn could you backport this to 0.64?

Sure. PR goes out today.

**Pull request merged**